src/clj/rems/api/applications_v2.clj

```diff
@@ -0,0 +1,810 @@
+                            :field/type (keyword (:type item))
+                            :field/title (localization-for :title item)
+                            :field/placeholder (localization-for :inputprompt item)
+                            :field/optional (:optional item)
+                            :field/options (:options item)
+                            :field/max-length (:maxlength item)})
+                         (:items form))
+        fields (merge-lists-by :field/id rich-fields app-fields)
+        description (->> fields
+                         (filter #(= :description (:field/type %)))
+                         first
+                         :field/value)]
+    (-> application
+        (assoc-in [:application/form :form/title] (:title form))
+        (assoc-in [:application/form :form/fields] fields)
+        (assoc :application/description description))))
+
+(defn- enrich-resources [app-resources get-catalogue-item]
+  (->> app-resources
+       (map :catalogue-item/id)
+       (map get-catalogue-item)
+       (map (fn [item]
+              {:catalogue-item/id (:id item)
+               :resource/id (:resource-id item)
+               :resource/ext-id (:resid item)
+               :catalogue-item/title (assoc (localization-for :title item)
+                                            :default (:title item))
+               :catalogue-item/start (:start item)
+               :catalogue-item/state (keyword (:state item))}))
+       (sort-by :catalogue-item/id)))
+
+(defn- enrich-licenses [app-licenses get-license]
+  (let [rich-licenses (->> app-licenses
+                           (map :license/id)
+                           (map get-license)
+                           (map (fn [license]
+                                  (let [type (keyword (:licensetype license))
+                                        content-key (case type
+                                                      :link :license/link
+                                                      :text :license/text)]
+                                    {:license/id (:id license)
+                                     :license/type type
+                                     :license/start (:start license)
+                                     :license/end (:end license)
+                                     :license/title (assoc (localization-for :title license)
+                                                           :default (:title license))
+                                     content-key (assoc (localization-for :textcontent license)
+                                                        :default (:textcontent license))})))
+                           (sort-by :license/id))]
+    (merge-lists-by :license/id rich-licenses app-licenses)))
+
+(defn- assoc-injections [application {:keys [get-form get-catalogue-item get-license get-user]}]
+  (-> application
+      (enrich-form get-form)
```

Comment:
I'd prefer `enrich-form` and `enrich-resources`, ... to have the same shape.

Reply:
Done. `set-application-description` is the exception, because it depends on the result of `enrich-form` but updates a key outside `:application/form`.